File: src/main/java/trivia/Player.java
package trivia;

public class Player {
    private final String name;
    private int position = 1;
    private int coins = 0;
    private boolean inPenaltyBox = false;

    public Player(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    public int getCoins() {
        return coins;
    }

    public void addCoin() {
        this.coins++;
    }

    public void advanceBy(int roll, int boardSize) {
        this.position += roll;
        if (this.position > boardSize) {
            this.position -= boardSize;
        }
    }

    public boolean hasWon(int winningCoins) {
        return this.coins == winningCoins;
    }

    public boolean isInPenaltyBox() {
        return inPenaltyBox;
    }

    public void setInPenaltyBox(boolean inPenaltyBox) {
        this.inPenaltyBox = inPenaltyBox;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: python/player.py
class Player:
    def __init__(self, name: str):
        self.name = name
        self.position = 1
        self.coins = 0
        self.in_penalty_box = False

    def add_coin(self):
        self.coins += 1

    def advance_by(self, roll: int, board_size: int):
        self.position += roll
        if self.position > board_size:
            self.position -= board_size

    def has_won(self, winning_coins: int) -> bool:
        return self.coins == winning_coins

    def __str__(self):
        return self.name
